torch/_dynamo/variables/builder.py

```diff
@@ -686,6 +686,41 @@ def index_source(key):
                 source=self.source,
                 guards=make_guards(GuardBuilder.FUNCTION_MATCH),
             )
+        elif isinstance(value, torch.SymInt):
```

Comment:
This seems incomplete? I was expecting you to accept torch.SymBool directly in here, and deal with the int-ify/un-int-ify transparently.

Reply:
Yeah, the current implementation is a blindfolded implementation of "need a conversion of the SymBool into a SymInt before it hits the graph".  The graph I understand is dynamo's compiled region. To achieve that, I come up with the idea of  doing the conversion in cond so that both user and dynamo don't need to worry about the conversion.

If we were to detect SymBool then converting it to SymInt inside dynamo, the SymBool inputs will hit the graph. I can explore the implementation a bit.